offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/offer/create/CreateOfferViewModel.groovy

```diff
@@ -152,8 +152,23 @@ class CreateOfferViewModel {
         for (ProductItemViewModel currentItem : alreadyExistingItems) {
             totalAmount = totalAmount + currentItem.quantity
         }
-        productItems.add(new ProductItemViewModel(totalAmount, item.product))
         productItems.removeAll(alreadyExistingItems)
+        productItems.add(new ProductItemViewModel(totalAmount, item.product))
+    }
+
+    void updateItem(ProductItemViewModel item) {
+        if (item.quantity <= 0.0) {
+            productItems.remove(item)
+            setChanged()
+            notifyObservers()
+            return
+        }
+        ProductItemViewModel existingItem = productItems.find { it.product.productId.equals(item.product.productId) } as ProductItemViewModel
+        int index = productItems.findIndexOf { it.product.productId.equals(item.product.productId) }
```

Comment:
you use a lot of implicit groovy magical casting here.
This will not be checked by the compiler and is prone to break.
Maybe consider rewriting this logic using java?